utils/src/lib.rs

```diff
@@ -145,11 +149,18 @@ pub fn link_with_nonce_or_address<T, U>(
         .iter()
         .map(|(id, contract)| {
             let key = id.identifier();
+            let version = id.version.to_string();
+            let version =
+                &version[..version.find("+").expect("Should have compiler version")].to_string();
```

Comment:
is this always guaranteed to have metadata?

Reply:
Yep, all solidity compiler versions have the metadata appended to them, but we can avoid this panic